packages/exhibitor/extra/dcos_zk_backup.py

```diff
@@ -0,0 +1,322 @@
+        parser = ArgumentParser(
+            description=(
+                'Create a backup of the ZooKeeper instance running on this '
+                'DC/OS master node'
+            )
+        )
+        parser.add_argument(
+            "backup_path",
+            type=_non_existing_file_path_existing_parent_dir,
+            help="File path that the gzipped ZooKeeper backup tar archive will be written to",
+        )
+        parser.add_argument(
+            "-t", "--tmp-dir",
+            type=_existing_dir_path,
+            help="Location of an existing directory to be used as temporary directory in the process",
+        )
+        parser.add_argument(
+            "-v", "--verbose",
+            action='store_true',
+            help="Display the output of every command",
+        )
+        args = parser.parse_args(sys.argv[2:])
+        if args.tmp_dir is None:
+            with TemporaryDirectory(suffix='-zk-backup') as tmp_dir:
+                backup_zookeeper(
+                    backup=args.backup_path,
+                    tmp_dir=Path(tmp_dir),
+                    verbose=args.verbose,
+                )
+        else:
+            backup_zookeeper(
+                backup=args.backup_path,
+                tmp_dir=args.tmp_dir,
+                verbose=args.verbose,
+            )
+
+    def restore(self) -> None:
+        """
+        Procedure invoked on `restore` command.
+        """
+        parser = ArgumentParser(
+            description=(
+                'Restore the ZooKeeper instance running on this DC/OS master '
+                'node from the given backup'
+            ),
+        )
+        parser.add_argument(
+            "backup_path",
+            type=_existing_file_path,
+            help="File path to the gzipped ZooKeeper backup tar archive to restore from",
+        )
+        parser.add_argument(
+            "-t", "--tmp-dir",
+            type=_existing_dir_path,
+            help="Location of an existing directory to be used as temporary directory in the process",
+        )
+        parser.add_argument(
+            "-v", "--verbose",
+            action='store_true',
+            help="Display the output of every command",
+        )
+        args = parser.parse_args(sys.argv[2:])
+        if args.tmp_dir is None:
+            with TemporaryDirectory(suffix='-zk-restore') as tmp_dir:
+                restore_zookeeper(
+                    backup=args.backup_path,
+                    tmp_dir=Path(tmp_dir),
+                    verbose=args.verbose,
+                )
+        else:
+            restore_zookeeper(
+                backup=args.backup_path,
+                tmp_dir=args.tmp_dir,
+                verbose=args.verbose,
+            )
+
+
+if __name__ == "__main__":
+    try:
+        DCOSZooKeeperCli()
+    except subprocess.CalledProcessError as exc:
+        if exc.output:
+            sys.stdout.buffer.write(exc.output)
+        sys.exit(exc.returncode)
+
+# vi: ft=python
```

Comment:
Is this line necessary?

Reply:
Good catch, it is no longer necessary because `dcos_zk_backup.py` is renamed in the `build` script so that `tox` recognizes it as a Python script and runs linting on it. Before it was `dcos_zk_backup` without `.py`, then this line was essential to me :)